comptests/s3/test_rest_api_rgw.py

```diff
@@ -282,3 +284,339 @@ def test_user_create_36665(self):
+        assert status[0] == HTTPStatus.NOT_FOUND, "User policy deleted"
+        self.log.info("Step 7: Verified test to remove invalid policy name")
+        self.log.info("END : test to validate delete invalid policy")
+        self.created_users.append(user_params)
+        self.log.info("END: %s",test_case_name)
+    # pylint: disable=C0301
+    @pytest.mark.api_user_ops
+    @pytest.mark.tags('TEST-41414')
+    def test_41414(self):
+        """Test to delete policy by using user Access key and secret key"""
+        self.log.info("Step 1: Create IAM user")
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        user_name = f"{self.user_name_prefix}{str(time.perf_counter_ns()).replace('.', '_')}"
+        email=f"{user_name}@seagate.com"
+        user_params = {
+            'display-name': user_name,
+            'email' : email,
+            'uid' : user_name
+        }
+        self.log.info("Step 2: Started creating IAM user.")
+        loop = asyncio.get_event_loop()
+        status, user_info = loop.run_until_complete(self.obj.create_user(user_params))
+        self.log.info(user_info)
+        access_key = user_info['keys'][0]['access_key']
+        secret_key = user_info['keys'][0]['secret_key']
+        self.log.info(
+            "Step 3: IAM user is created successfully")
+        policy = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Action": "iam:DeleteUserPolicy",
+                    "Resource": "arn:aws:iam:::user/"+f"{user_name}"
+                }
+            ]
+        }
+        self.log.info(
+            "Step 4: Applying user policy to the user")
+        self.log.info(policy)
+        policy_document = json.dumps(policy)
+        user_params2 = {
+            'UserName' : user_name,
+            'PolicyName' : 'Policy1',
+            'PolicyDocument' : policy_document,
+            'Action' : 'PutUserPolicy',
+            'format' : 'json'
+        }
+        loop = asyncio.get_event_loop()
+        status = loop.run_until_complete(self.obj.put_user_policy(user_params2))
+        self.log.info(
+            "Step 5: Successfully applied policy to the user")
+        self.log.info("Step 6: Deleting Policy by using user Access key and Secret key")
+        loop = asyncio.get_event_loop()
+        user_params3 = {
+            'UserName' : user_name,
+            'PolicyName' : 'Policy1',
+            'Action' : 'DeleteUserPolicy',
+            'format' : 'json'
+        }
+        self.log.info("Step 7: Deleting Policy by using user Access key and Secret key")
+        status = loop.run_until_complete(self.obj.delete_policy_with_user_keys(user_params3, access_key, secret_key))
+        assert status[0] == HTTPStatus.OK, "Not able to delete policy"
+        self.log.info("Step 8: Successfully delete policy by self")
```

Comment:
Please verify policy got deleted by using list user policy/get user policy

Reply:
Here, first verifying the delete user policy by self, once the policy is deleted user can't access to get user policy , in policy it self we have delete and get user policy access

Reply:
yes, currently you relying on status of delete policy command.
where as you can confirm that with list (it should not list deleted policy) or get it will give 404 error
its just a crosscheck.